File: program136.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj=new Scanner(System.in);
        System.out.println("Enter number");
        this.iNo=sobj.nextInt();

    }

    public void Display()
    {
        System.out.println("Value is:"+this.iNo);

    }

    public int Factorial()
    {
        int iFact=1;
     for(int i=1;i<=iNo;i++)
    {
        iFact=iFact*i;       //iFact*=i;(shorthand assignment operator)
    }
     return iFact;

    }
}

class program136
{
    public static void main(String arg[])
    {
        Number nobj=new Number();
        int iRet=0;

        nobj.Accept();
        nobj.Display();;

        iRet=nobj.Factorial();
        System.out.println("factorial is:"+iRet);
    }
}

File: program136.cpp
//factorial in c++ best code
//4
//4*3*2*1=24
//1*2*3*4=24
#include<iostream>
using namespace std;

class Number
{
    private:
    int iNo;    //characteristic

    public:
    //behaviours
    void Accept()   //setter method
    {
        cout<<"Enter the value:"<<endl;
        cin>>this->iNo;
    }

    void Display()   //getter method
    {
        cout<<"Value is :"<<this->iNo<<endl;
    }

    int Factorial()
{  
    int iFact=1;
    for(int i=1;i<=iNo;i++)
    {
        iFact=iFact*i;       //iFact*=i;(shorthand assignment operator)
    }
    return iFact;

}
};


int main()
{
    Number nobj;
    int iRet=0;
    
    //cout<<nobj.iNo;  private 
    nobj.Accept();
    nobj.Display();

    iRet=nobj.Factorial();
    cout<<"factorial is:"<<iRet<<endl;
    return 0;
}